File: models/Venta.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Venta = sequelize.define('venta', {
  id_venta: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
    field: 'id_venta'
  },
  cantidad: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  precio_unitario: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'precio_unitario'
  },
  precio_total: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'precio_total'
  },
  fecha: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  },
  estado: {
    type: DataTypes.ENUM('completada', 'cancelada', 'devuelta'),
    defaultValue: 'completada'
  },
  cancelado_por: {
    type: DataTypes.INTEGER
  },
  fecha_cancelacion: {
    type: DataTypes.DATE
  },
  motivo_cancelacion: {
    type: DataTypes.STRING
  },
  producto_id: {
    type: DataTypes.INTEGER,
    references: {
      model: 'productos', // Nombre de la tabla (no del modelo)
      key: 'id_producto'
    }
  }
}, {
  tableName: 'ventas',
  timestamps: true,
  underscored: true
});

// No establecer relaciones aquí - se hará en app.js
module.exports = Venta;

File: lib/models/venta.dart
class Venta {
  final int idVenta;
  final int idProducto;
  final String nombreProducto;
  final int cantidad;
  final double precioUnitario;
  final double precioTotal;
  final DateTime fecha;
  final int idVendedor;
  final String nombreVendedor;
  final String estado; // 'completada', 'cancelada', 'devuelta'
  final int? idSupervisorCancelacion;
  final DateTime? fechaCancelacion;
  final String? motivoCancelacion;

  Venta({
    required this.idVenta,
    required this.idProducto,
    required this.nombreProducto,
    required this.cantidad,
    required this.precioUnitario,
    required this.precioTotal,
    required this.fecha,
    required this.idVendedor,
    required this.nombreVendedor,
    this.estado = 'completada',
    this.idSupervisorCancelacion,
    this.fechaCancelacion,
    this.motivoCancelacion,
  });

  factory Venta.fromJson(Map<String, dynamic> json) {
    return Venta(
      idVenta: json['idVenta'],
      idProducto: json['idProducto'],
      nombreProducto: json['nombreProducto'],
      cantidad: json['cantidad'],
      precioUnitario: json['precioUnitario'].toDouble(),
      precioTotal: json['precioTotal'].toDouble(),
      fecha: DateTime.parse(json['fecha']),
      idVendedor: json['idVendedor'],
      nombreVendedor: json['nombreVendedor'],
      estado: json['estado'],
      idSupervisorCancelacion: json['idSupervisorCancelacion'],
      fechaCancelacion: json['fechaCancelacion'] != null 
          ? DateTime.parse(json['fechaCancelacion']) 
          : null,
      motivoCancelacion: json['motivoCancelacion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idVenta': idVenta,
      'idProducto': idProducto,
      'nombreProducto': nombreProducto,
      'cantidad': cantidad,
      'precioUnitario': precioUnitario,
      'precioTotal': precioTotal,
      'fecha': fecha.toIso8601String(),
      'idVendedor': idVendedor,
      'nombreVendedor': nombreVendedor,
      'estado': estado,
      'idSupervisorCancelacion': idSupervisorCancelacion,
      'fechaCancelacion': fechaCancelacion?.toIso8601String(),
      'motivoCancelacion': motivoCancelacion,
    };
  }
}